samples/pluggable_allocator/enclave/Makefile

```diff
@@ -33,8 +34,8 @@ build:
 	oeedger8r ../allocator_demo.edl --trusted \
 		--search-path $(INCDIR) \
 		--search-path $(INCDIR)/openenclave/edl/sgx
-	$(CXX) -g -c $(CXXFLAGS) -DOE_API_VERSION=2 enc.cpp -o enc.o
-	$(CC) -g -c $(CFLAGS) -DOE_API_VERSION=2 allocator_demo_t.c -o allocator_demo_t.o
+	$(CXX) -g -c $(CRYPTO_FLAGS) $(CXXFLAGS) -DOE_API_VERSION=2 enc.cpp -o enc.o
```

Comment:
One issue - this is a breaking change to existing apps. We should flag it as such! They would all definitely need to change it. I wonder if there is any way around this

Reply:
"CRYPTO_FLAGS" is an empty string for all crypto libs except openssl_3. Existing apps can continue building with openssl/mbedtls/etc and ignoring CRYPTO_FLAGS and should not be affected.

But this still may be confusing and/or not clearly presented, and I'm open to suggestions on a more intuitive or better explained way.

Reply:
Is there a way to push CRYPTO_FLAGS into  $(CXXFLAGS)(or CFLAGS)?
Like CXXFLAGS = CXXFLAGS + CRYPTO_FLAGS somewhere?

Reply:
We can absolutely do CXXFLAGS = CRYPTO_FLAGS + CXXFLAGS in the sample Makefiles. I'll make that change.

The problem is that, regardless, customers will need to pull "--variable=${OE_CRYPTO_LIB}flags" info from pkg-config to build with openssl_3. We can't add the additional includes to pkg-config --cflags because they are different for openssl_3 vs openssl

Reply:
Could we put together a guidance for customers upgrading to OpenSSL 3? And, Is there anything customers need to do if not upgrading?

Reply:
To you have advise where to put that guidance? Maybe a new section in docs/OpenSSLSupport.md?

The way this current change is now, customers who don't upgrade should not have to do anything.

Reply:
I reformatted the Makefiles to concatenate CRYPTO_FLAGS with CFLAGS. I also added a new section in docs/OpenSSLSupport.md with some pointers on building enclaves with OpenSSL 3.0.